src/RecordSession.cc

```diff
@@ -2230,6 +2230,10 @@ static string lookup_by_path(const string& name) {
     env.push_back("SYSTEMD_RDRAND=0");
   }
 
+#if defined(__SANITIZE_ADDRESS__)
+  env.push_back("ASAN_OPTIONS=verify_asan_link_order=0");
```

Comment:
I've been thinking about this, and I'd suggest that this should be fine as a testsuite-specific hack provided there's a comment mentioning that
1. This is needed to run the test suite with an ASan-instrumented rr because of rr's lack of support for children `exec`ing ASan binaries, and should be removed if/when rr gains such support.
2. This is specific to the case of running the test suite and isn't supposed to help with general usage.

Please feel free to word it slightly less awkwardly ;)

Reply:
Going to add a comment.